engine/schema/src/com/cloud/configuration/dao/ResourceCountDaoImpl.java

```diff
@@ -276,17 +276,18 @@ public long countMemoryAllocatedToAccount(long accountId) {
     }
 
     private long executeSqlCountComputingResourcesForAccount(long accountId, String sqlCountComputingResourcesAllocatedToAccount) {
-        try (TransactionLegacy tx = TransactionLegacy.currentTxn()) {
+        TransactionLegacy tx = TransactionLegacy.currentTxn();
+        try {
             PreparedStatement pstmt = tx.prepareAutoCloseStatement(sqlCountComputingResourcesAllocatedToAccount);
             pstmt.setLong(1, accountId);
 
             ResultSet rs = pstmt.executeQuery();
             if (!rs.next()) {
-              throw new CloudRuntimeException(String.format("An unexpected case happened while counting allocated computing resources for account: " + accountId));
+                return 0L;
             }
             return rs.getLong("total");
         } catch (SQLException e) {
             throw new CloudRuntimeException(e);
```

Comment:
What about adding a message similar to the "An unexpected case happened while counting allocated..." to the exception stack?

Reply:
@GabrielBrascher actually it is not an error. rs.next() is false means the sql query returns null (for example, no vm in the domain/account). If there are something wrong in the sql query, they are handled in line 290.

Reply:
@ustcweizhou I was referring to the exception "CloudRuntimeException(e)" at line 290. Considering an exception "user-friendly". But it is a minor detail, overall it LGTM.

Ex: `throw new CloudRuntimeException(String.format("Failed to count allocated computing resources for account [id:%s]", accountId), e);`.